### PR TITLE
Add snapshot management

### PR DESCRIPTION
**Pull request by emerkle826**
This PR adds GET, DELETE, and POST methods for retrieving, clearing and taking snapshots, respectively. This should resolve [issue #52](https://github.com/datastax/management-api-for-apache-cassandra/issues/52)

### CODE REVIEW

**Comment by emerkle826** [comment]
@tjake This should provide most of the same capabilities as `nodetool snapshot`, `nodetool clearsnapshot`, and `nodetool listsnapshots`

**Comment by emerkle826** [comment]
The 2 failing tests in the checks are due to tests for truncating Hints against Cassandra 4.0. I've tried to address that in a different PR [here](https://github.com/datastax/management-api-for-apache-cassandra/pull/54)

File: management-api-server/src/test/java/com/datastax/mgmtapi/NonDestructiveOpsIntegrationTest.java [hunk 1]
```diff
@@ -464,6 +469,64 @@ public void testAlterKeyspace() throws IOException, URISyntaxException
...
+
+		boolean takeSnapshotSuccessful = client.post(takeSnapshotUri.toURL(), requestAsJSON).thenApply(r -> r.status().code() == HttpStatus.SC_OK).join();
+		assertTrue(takeSnapshotSuccessful);
+
+		// get snapshot details
+		URI getSnapshotsUri = uriBuilder.addParameter("snapshotNames", "testSnapshot").build();
+		String getSnapshotResponse = client.get(getSnapshotsUri.toURL())
+				.thenApply(this::responseAsString).join();
+		assertNotNull(getSnapshotResponse);
+        Object responseObject = ReaderUtility.read(Object.class, MediaType.APPLICATION_JSON, getSnapshotResponse);
```
Comment (tjake) [inline management-api-server/src/test/java/com/datastax/mgmtapi/NonDestructiveOpsIntegrationTest.java:495]: Formatting

**Review by tjake — CHANGES_REQUESTED** [review]
Great stuff, could you rebase and fix a formatting nit? Then good to go

**Review by tjake — APPROVED** [review]

**Pull request merged**